WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -113,7 +113,7 @@ def update
   end
 
   private def redirect_if_cannot_edit_user(user)
-    unless current_user && (current_user.can_edit_users? || current_user == user)
+    if !current_user&.can_edit_user?(user)
```

Comment:
As you reduced it to a single condition using `!`, I'd stick with `unless`, but it's a personal thing, so feel free to ignore =)